### DISCUSSION TITLE
Build failed with 0 errors on dotnet 7 rc (non-deterministically)

### DISCUSSION THREAD

**Discussion by Shayan-To** [dotnet/runtime#76091] [Q&A]
I have an almost-empty asp.net webapi project created by dotnet 7 rc, and when trying to build/run it, sometimes it fails with no errors and no warnings. It sometimes does this and sometimes doesn't, seems completely non-deterministic. I tried `-v diag`, and it shows me this:

```
PS> dotnet run -v diag  
Using launch settings from /home/me/Codes/Test-Project/Properties/launchSettings.json...
Building...
/home/me/dotnet-sdk-7.0.100-rc.1.22431.12-linux-x64/sdk/7.0.100-rc.1.22431.12/MSBuild.dll -nologo -maxcpucount -restore -verbosity:m -verbosity:diag /home/me/Codes/Test-Project/Test-Project.csproj
Build started 9/23/2022 9:52:36 PM.
21:52:36.215     0>Process = "/home/me/dotnet/dotnet-sdk-7.0.100-rc.1.22431.12-linux-x64/dotnet"
                   MSBuild executable path = "/home/me/dotnet/dotnet-sdk-7.0.100-rc.1.22431.12-linux-x64/sdk/7.0.100-rc.1.22431.12/MSBuild.dll"
                   Command line arguments = "/home/me/dotnet/dotnet-sdk-7.0.100-rc.1.22431.12-linux-x64/sdk/7.0.100-rc.1.22431.12/MSBuild.dll -maxcpucount -verbosity:m -restore /home/me/Codes/Test-Project/Test-Project.csproj -nologo -verbosity:diag"
                   Current directory = "/home/me/Codes/Test-Project"
                   MSBuild version = "17.4.0-preview-22428-01+14c24b2d3"

Build FAILED.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.01

The build failed. Fix the build errors and run again.
```

When I run the exact command that is shown above (that is, I run MSBuild directly), it works fine every time and everything is okay.

(Ubuntu 20.04 with all apt upgrades, dotnet sdk 7.0.100-rc.1.22431.12 for linux x64)

**Comment by rokonec** [comment]
@Shayan-To Can you please somehow share this Test-Project with me, I would like to use if for repro.
Or you can set `export MSBuildDebugEngine=1` repro it and then share with me content of `.\MSBuild_Logs` folder? However content of that folder might contains private and sensitive data and I would not recommend to share it publicly. Best approach is to create VS feedback, as described [here](https://learn.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022) and share the link to it with me.

**Comment by Shayan-To** [comment]
Here's the logs: (Checked it for private info.)
[MSBuild_CommTrace_PID_65834.txt](https://github.com/dotnet/runtime/files/9636601/MSBuild_CommTrace_PID_65834.txt)

I had a new observation trying to get it to fail. I run `dotnet run` and it works fine. Then I run `dotnet watch` and it fails. Then I run `dotnet run` again, and it fails, even in the subsequent runs. (I don't know what I had done that `dotnet run` was working at the beginning.)

**Reply by rokonec** [reply]
Because there were probably persitent process, build server, which started before you have set that env, we need to kill it. 
Please run `pkill dotnet && rm -rf MSBuild_Logs`,  repro again, and upload all MSBuild_CommTrace_PID_*.txt . Those particular logs rarely contains sensitive data, but binlog does.

**Reply by Shayan-To** [reply]
https://developercommunity.visualstudio.com/t/Build-failed-with-0-errors-on-dotnet-7/10156587?space=61&ftype=problem&preview2=true&entry=problem

BTW, when I run `pkill dotnet`, then `dotnet run` works fine. I ran `dotnet watch` for it to fail.

**Reply by Shayan-To** [reply]
Is there anything else I can do to help with this?

**Reply by rokonec** [reply]
By looking on the content of the files, it looks like the build have had succeeded and that you zipped it before it start failing.

I think you deserve more context about what is going on:
In .net 7 we have introduced concept of build server. It is persistent process which can reuse cached memory data from previous builds. After 15 minutes of idling, build server shut down on its own.
It looks like in your case 1st execution of build in just created build server works but subsequent leveraging already running build server fails.
I am very happy you cooperate with hunting this bug. To understand this failure better I need MSBuild_Logs after it failed. Or if you can share with me your solution so I can repro and debug it locally.

**Reply by rokonec** [reply]
@Shayan-To Thanks to your help, we managed to repro it with our own basic ASP project. For now on, we do not need any more traces from your side.
We will keep you informed.

If this issue is blocking you there is workaround, turn off build server by setting the environment variable `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER` to `1` by `export DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER=1` command.

Thank you for your cooperation, it was very helpful.

**Reply by Shayan-To** [reply]
Thanks!

I was trying different things to be able to produce better logs, or a minimal repro. I'm glad that you could repro it yourself.

I hope the problem gets fixed soon.

**Reply by rokonec** [reply]
It has been fixed in: https://github.com/dotnet/msbuild/pull/8006
It might take few days for this fix to make it into SDK
